### ISSUE TITLE
songArt1 != songArt2 (when both songs are in the same album and have the same image)

### ISSUE THREAD

**Issue by shaan-mephobic** [LucasPJS/on_audio_query#59]
I'm tryna cache all the artworks for a better performance. Although some albums have different artwork for each song, there are albums that have the same artwork for every song. If there was a way to figure out if the whole album uses the same image, I could save up a lot of space by caching only one of those images. I tried to check if the `Uint8Lists` are equal for 2 different songs with same artwork in the same album but it's always false. I assume the compression is random for every image so it returns false, so a property **like** `isOnlyArtwork` to see if the whole album has only one image would save up a lot of space.

TL;DR - A way to find out if a whole album has only one artwork

**Comment by LucasPJS** [comment]
Using the `==`, dart will compare a specific type(idk why or how this happen). In your case, you need to compare every single image byte.

Try using [this](https://stackoverflow.com/questions/10404516/how-can-i-compare-lists-for-equality-in-dart/55974120#55974120) method.

```dart
import 'package:flutter/foundation.dart';

Uint8List art1 = ...;
Uint8List list2 = ...;

debugPrint(listEquals(art1, art2));
```

**Comment by shaan-mephobic** [comment]
That worked! Thanks :)
